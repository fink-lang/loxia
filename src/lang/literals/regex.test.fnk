{fink2js} = import '../../testing'
{describe, it, expect, to_match_snapshot} = import '@fink/jest'


describe:: 'regex', fn:
  it:: 'compiles single line', fn:
    expect::
      fink2js`
        regex = rx/(?<year>\\d{4})-(?<month>\\d{2})-(?<day>\\d{2})/
      `
      to_match_snapshot


  it:: 'compiles multiline', fn:
    expect::
      fink2js`
        regex = rx/
          (?<year>\\d{4})-    # year part of a date
          (?<month>\\d{2})-   # month part of a date
          (?<day>\\d{2})      # day part of a date
        /gm
      `
      to_match_snapshot


  it:: 'compiles escape char', fn:
    expect::
      fink2js`
        regex = rx/.+\\/.+\\\\/
      `
      to_match_snapshot


