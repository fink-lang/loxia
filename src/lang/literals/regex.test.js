import {fink2js} from '../../testing';


describe('regex', ()=> {
  it('compiles single line', ()=> {
    expect(
      fink2js(`
        regex = rx/(?<year>\\d{4})-(?<month>\\d{2})-(?<day>\\d{2})/
      `)
    ).toMatchSnapshot();
  });


  it('compiles multiline', ()=> {
    expect(
      fink2js(`
        regex = rx/
          (?<year>\\d{4})-    # year part of a date
          (?<month>\\d{2})-   # month part of a date
          (?<day>\\d{2})      # day part of a date
        /gm
      `)
    ).toMatchSnapshot();
  });


  it('compiles escape char', ()=> {
    expect(
      fink2js(`
        regex = rx/.+\\/.+\\\\/
      `)
    ).toMatchSnapshot();
  });
});
