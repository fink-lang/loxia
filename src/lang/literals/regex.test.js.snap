// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiles regex 1`] = `
"const regex1 = /(?<year>\\\\d{4})-(?<month>\\\\d{2})-(?<day>\\\\d{2})/gm;
const regex2 = /(?<year>\\\\d{4})-(?<month>\\\\d{2})-(?<day>\\\\d{2})/;
const regex3 = /.+\\\\/.+/;
Object.assign(module.exports, {
  regex1,
  regex2,
  regex3
});"
`;
